### PR TITLE
Adds an authorized cookie

### LINKED ISSUE #215: The server should grant on-campus access to an off-campus client that has previously been on campus
Enable this scenario:
1. A user U visits ABE via a browser B on device D on the campus network.
2. U visits ABE via B/D from off campus.
3. U should see non-public events, and have edit permission on events.

These things are not requirements:
* The server can revoke or audit auth grants.
* Auth grants expire.
* U can revoke grants from their browser or device (although clearing cookies and/or local storage may do this, depending on the implementation)

A possible strategies:
1. When the device is on campus, the server sends back a cookie that include a grant. This could just be a cookie whose presence is tested, for now.
2. The server code that currently checks whether the client is inside the LAN, is modified to also test whether this cookie is present.

Variants, that enable non-required features and security:
* The cookie value is a shared secret. This makes it more difficult to programmatically forge access.
* The cookie value includes the issue date (or, an expiration date). This allows us to expire grants.
* The cookie value is a JWT or other encrypted secret. This prevents the client from tampering with the expiration date, or extracting the plaintext of the secret.

### PR DESCRIPTION
**Pull request by jaredbriskman**
Closes #215 .

After discussion, decided to implement a cookie with a shared secret for simplicity.
Currently, there is no expiry or timestamp on the cookie, although this could be added.
I think this all will change once we're able to move to a security domain model with users though.

Going to bring up a review app for a little more testing first.

(Also,tried to upgrade to flask 1.0, now #221 )

### CODE REVIEW

**Comment by jaredbriskman** [comment]
The review app behaves as expected too. When this is merged, the environment variable `SHARED_SECRET` may need to be set on existing heroku apps.

**Comment by jaredbriskman** [comment]
After discussion with @songbird175 , added c78d26c to help with #103 . Now the auth check is a separate function from the auth-requried decorator, which should make integration with more nuanced behavior than throwing 401 errors simpler, while still setting the Auth cookie if the check is passed.

File: tests/test_auth.py [hunk 1]
```diff
@@ -76,5 +75,15 @@ def route():
             with self.assertRaises(HTTPException) as http_error:
                 route()
 
-if __name__ == '__main__':
-    unittest.main()
+        # Test auth cookie
+        os.environ['SHARED_SECRET'] = 'security'
+        auth = reload(auth)
+
+        with app.test_request_context('/', headers={"COOKIE": "app_secret=security"},
```
Comment (osteele) [inline tests/test_auth.py]: I suggest wrapping this and the next `with app.test_request_context` within separate `with self.subTest(…):`s, for better test reporting.

Comment (jaredbriskman) [reply tests/test_auth.py]: Thanks for the suggestion, implemented in 4b35300 !

**Pull request merged**